src/dlgIRC.cpp

```diff
@@ -20,306 +22,715 @@
+    QShortcut* shortcut = new QShortcut(Qt::Key_Tab, this);
+    QShortcut* shortcut2 = new QShortcut(Qt::Key_Up, this);
+    connect(shortcut, &QShortcut::activated, this, &dlgIRC::slot_nameCompletion);
+    connect(shortcut2, &QShortcut::activated, this, &dlgIRC::slot_onHistoryCompletion);
+
+
+    connect(lineEdit, &QLineEdit::returnPressed, this, &dlgIRC::slot_onTextEntered);
+    connect(lineEdit, &QLineEdit::textEdited, this, &dlgIRC::slot_onTextEdited);
+    connect(ircBrowser, &QTextBrowser::anchorClicked, this, &dlgIRC::slot_onAnchorClicked);
+    connect(userList, &QListView::doubleClicked, this, &dlgIRC::slot_onUserActivated);
+    connect(connection, &IrcConnection::connected, this, &dlgIRC::slot_onConnected);
+    connect(connection, &IrcConnection::connecting, this, &dlgIRC::slot_onConnecting);
+    connect(connection, &IrcConnection::disconnected, this, &dlgIRC::slot_onDisconnected);
+    connect(connection, &IrcConnection::nickNameRequired, this, &dlgIRC::slot_nickNameRequired);
+    connect(connection, &IrcConnection::nickNameChanged, this, &dlgIRC::slot_nickNameChanged);
+    connect(connection, &IrcConnection::joinMessageReceived, this, &dlgIRC::slot_joinedChannel);
+    connect(connection, &IrcConnection::partMessageReceived, this, &dlgIRC::slot_partedChannel);
+    connect(connection, &IrcConnection::numericMessageReceived, this, &dlgIRC::slot_receiveNumericMessage);
+
+    mUserName = QStringLiteral("mudlet");
+    mRealName = mudlet::self()->version;
+    mHostName = readIrcHostName(mpHost);
+    mHostPort = readIrcHostPort(mpHost);
+    mNickName = readIrcNickName(mpHost);
+    mChannels = readIrcChannels(mpHost);
+
+    connection->setNickName(mNickName);
+    connection->setUserName(mUserName);
+    connection->setRealName(mRealName);
+    connection->setHost(mHostName);
+    connection->setPort(mHostPort);
+}
 
-    if (nick.isEmpty()) {
-        nick = tr("Mudlet%1").arg(QString::number(rand() % 10000));
-        QFile file(QDir::homePath() + "/.config/mudlet/irc_nick");
-        file.open(QIODevice::WriteOnly | QIODevice::Unbuffered);
-        QDataStream ofs(&file);
-        ofs << nick;
-        file.close();
+dlgIRC::~dlgIRC()
+{
+    if (connection->isActive()) {
+        const QString quitMsg = tr("%1 closed their client.").arg(mNickName);
+        connection->quit(quitMsg);
+        connection->close();
+    }
+}
+
+void dlgIRC::startClient()
+{
+    if (mIrcStarted) {
+        return;
     }
 
-    connection->setNickName(nick);
-    mNick = nick;
-    connection->setUserName("mudlet");
-    connection->setRealName(mudlet::self()->version);
-    connection->setHost("irc.freenode.net");
-    connection->setPort(6667);
+    connection->sendCommand(IrcCommand::createJoin(mChannels));
     connection->open();
+
+    setupBuffers();
+
+    ircBrowser->append(IrcMessageFormatter::formatMessage(tr("$ Statring Mudlet IRC Client...")));
```

Comment:
_Though I suppose the original term_ might _refer to a somewhat dubious late-night (adult) mathematics student party group activity._ 😈